Decouple './client/multi-currency-setup'

lovo-h opened the pull request:
Fixes 108-gh-Automattic/woopayments.

## Changes proposed in this Pull Request

<!--
Title: A descriptive, yet concise, title.
-->

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

The changes in this PR continue the MCCY migration process to V2. More specifically, the MCCY setup is migrated to V2. However, this is an intermediary change that will depend on the changes from the following issues to be completed before it is fully functional:
* 106-gh-Automattic/woopayments.
* 107-gh-Automattic/woopayments.

## Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

**Confirm Full Migration of `./client/multi-currency-setup**
* Start by running the `npm run test:js` command and ensure all tests pass.
* Ensure all files in `./client/multi-currency-setup` were moved to `./multi-currency/setup`, properly disconnected from WooPayments, properly reconnected into MCCY, and interfaced with WooPayments through the interface later (e.g. `./multi-currency/interface/*`). Note: there exist some dependencies to some UI components in `./client/multi-currency` that will need to be migrated. However, this cannot happen until after 107-gh-Automattic/woopayments has been merged.

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

lovo-h committed:
Move client/data/multi-currency/* to multi-currency/client/data/*

lovo-h committed:
Connect new MCCY data store: wc/payments/mccy

lovo-h committed:
Ensure non-MCCY reference the correct MCCY data files

lovo-h committed:
Remove unused MCCY test code

lovo-h committed:
WebPack config: Add 'multi-currency' dir and alias with 'mccy'

lovo-h committed:
Added a data interface layer where functions are shared

Sharing occurs between WooPayments v1 and MCCY v2.

lovo-h committed:
Move ./client/multi-currency-setup to ./multi-currency/client/setup

lovo-h committed:
Add interface layer: components, functions, & data

lovo-h committed:
Use the 'functions' interface to inject dependencies into MCCY

lovo-h committed:
Use the 'components' interface to inject dependencies into MCCY

lovo-h committed:
Use the 'data' interface to inject dependencies into MCCY

lovo-h committed:
Update multi-currency-setup's stale references to MCCY data files

lovo-h committed:
Export MultiCurrencySetupPage from MCCY to main codebase

lovo-h committed:
Add temporary dependency on MCCY + TODOs

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9348` or branch name `migrate/setup-multi-currency` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=migrate/setup-multi-currency) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 4996381ca00afc1f16747de6a9fc31664831c37f
- Build time: 2024-08-30 16:25:34 UTC

_Note: the build is updated when a new commit is pushed to this PR._

lovo-h committed:
Fix tests: useStoreSettings imported from client

As opposed to being imported from 'multi-currency/data'. So, we mock
'wcpay/data' in order to prevent the other store hooks/data from
initializing.

lovo-h committed:
Update 'mccy' alias to 'multi-currency'

lovo-h committed:
Merge branch 'migrate/data-multi-currency' into migrate/setup-multi-currency

lovo-h committed:
Update 'mccy' alias reference to 'multi-currency'

lovo-h committed:
Merge branch 'multi-currency-v2' into migrate/setup-multi-currency

ricardo commented on multi-currency/client/setup/tasks/add-currencies-task/index.js:
```diff
@@ -10,23 +10,26 @@ import _ from 'lodash';
+	Search,
+	LoadableBlock,
+} from 'multi-currency/interface/components';
 import WizardTaskItem from '../../wizard/task-item';
```
Question: It seems like we have a pretty similar `WizardTaskItem` from `interface/components`. I wonder if we should use only one? And adapt V1 to have a `visibleDescription` prop.

lovo-h replied:
Good call out. De-duplicated in [1424558](https://github.com/Automattic/woocommerce-payments/pull/9348/commits/142455874118f1a04069c2ed171c61de96a26bf8).

ricardo reviewed (commented):
@lovo-h I understand this also depends on 107-gh-Automattic/woopayments before it's fully functional, but should I run any specific tests at this point?

Left just one comment but overall the changes make sense and LGTM.

lovo-h committed:
De-duplicate WizardTaskItem

lovo-h committed:
Code cleanup: Place imports in correct section, fix typo

lovo-h committed:
Ensure getMultiCurrencyState refers to top level state

lovo-h commented:
> I understand this also depends on [107-gh-Automattic/woopayments](https://href.li/?https://github.com/Automattic/woopayments/issues/107) before it's fully functional, but should I run any specific tests at this point?

@ricardo: Actually, at this point, we can go ahead and test the multi-currency setup wizard:
* As a merchant, navigate to **WooCommerce > Home**.
* Scroll down to **Inbox** section.
* Click **Show older** button.
* Click the **Set up now** button in **Sell worldwide in multiple currencies**.
* Ensure currencies appear, can be selected, and that the merchant can still click the **Continue** button.
* Ensure `visibleDescription` is displayed in inactive sections:
  * When the **Add currencies** section is active, the **Review store settings** section should have the following subtext: "These settings can be changed any time by visiting the Multi-Currency settings"
  * Click the **Continue** button and ensure the **Add currencies** section has the following subtext: "# currencies added" (where # refers to the number of currencies added).

By running these tests, I ran into an issue where the currencies were not displaying. This was fixed in [4996381](https://github.com/Automattic/woocommerce-payments/pull/9348/commits/4996381ca00afc1f16747de6a9fc31664831c37f).

ricardo reviewed (approved):
LGTM and tests well.

- ✅ Multi-currency setup tests.
- ✅ Files were moved to `/multi-currency/client/setup`.
